api/v1/helpers/verbs/doGet.js

```diff
@@ -29,11 +30,33 @@ const httpStatus = require('../../constants').httpStatus;
  *  resource type to retrieve.
  */
 function doGet(req, res, next, props) {
-  u.findByKey(props, req.swagger.params)
-  .then((o) => {
-    res.status(httpStatus.OK).json(u.responsify(o, props, req.method));
-  })
-  .catch((err) => u.handleError(next, err, props.modelName));
+  if (props.cacheEnabled) {
+    redisCache.get(req.swagger.params.key.value, (cacheErr, reply) => {
```

Comment:
Yeah, I did not factor in the possibility of fields here. What if I create the cache key- a string which is combination of object and fields required. And I will remove the cache entry when some update happens, to keep it simple, so that it is fetched from db next time.